src/modules/hub.bicep

```diff
@@ -131,6 +145,207 @@ resource uploadSettingsJson 'Microsoft.Resources/deploymentScripts@2020-10-01' =
   }
 }
 
+// Key Vault for storing secrets
+
+module keyVault 'Microsoft.KeyVault/vaults/deploy.bicep' = {
+  name: 'keyVault'
+  dependsOn: [
+    dataFactory
+  ]
+  params: {
+    name: keyVaultName
+    location: location
+    tags: resourceTags
+    enablePurgeProtection: false
+    accessPolicies: [
+      {
+        objectId: dataFactory.outputs.systemAssignedPrincipalId
+        tenantId: subscription().tenantId
+        permissions: {
+          secrets: [
+            'get'
+          ]
+        }
+      }
+    ]
+  }
+}
+
+module keyVaultSecret 'secrets/storageaccountsecret.bicep' = {
+  name: 'keyVaultSecret'
+  dependsOn: [
+    storageAccount
+    keyVault
+  ]
+  params: {
+    keyVaultName: keyVaultName
+    storageAccountName: storageAccountName
+    secretName: storageAccountName
+  }
+}
+
+// Azure Data Factory and Pipelines
+module dataFactory 'Microsoft.DataFactory/factories/deploy.bicep' = {
+  name: 'dataFactory'
+  params: {
+    name: dataFactoryName
+    location: location
+    tags: resourceTags
+    systemAssignedIdentity: true
+  }
+}
+
+module linkedService_keyVault 'linkedservices/linkedservice.bicep' = {
+  name: 'linkedService_keyVault'
+  dependsOn: [
+    dataFactory
+    keyVaultSecret
+  ]
+  params: {
+    dataFactoryName: dataFactoryName
+    linkedServiceName: keyVaultName
+    linkedServiceType: 'AzureKeyVault'
+    linkedServiceTypeProperties:  {
+      baseUrl: keyVaultUri
+    }
+  }
+}
+
+module linkedService_storageAccount 'linkedservices/linkedservice.bicep' = {
+  name: 'linkedService_storageAccount'
+  dependsOn: [
+    dataFactory
+    linkedService_keyVault
+  ]
+  params: {
+    dataFactoryName: dataFactoryName
+    linkedServiceName: storageAccountName
+    linkedServiceType: 'AzureBlobFS'
+    linkedServiceTypeProperties: {
+      url: reference('Microsoft.Storage/storageAccounts/${storageAccountName}', '2019-04-01').primaryEndpoints.dfs
+      accountKey: {
+        type: 'AzureKeyVaultSecret'
+        store: {
+          referenceName: keyVaultName
+          type: 'LinkedServiceReference'
+        }
+        secretName: storageAccountName
+      }
+    }
+  }
+}
+
+module msexports 'datasets/adlsv2.bicep' = {
+  name: 'msexports'
```

Comment:
Should we call this out as a dataset? (Same for the others.)

Reply:
Done